Client/Frontend/Browser/SearchEngines.swift

```diff
@@ -51,12 +51,21 @@ class SearchEngines {
         }
 
         set(defaultEngine) {
+            let previousEngine = self.orderedEngines[0]
+            let changeKey = "BeforeCurrentWasDefault"
+            let shouldDisablePrevious = self.disabledEngineNames[changeKey] ?? false
+            if previousEngine.shortName != defaultEngine.shortName {
+                self.disabledEngineNames[changeKey] = self.disabledEngineNames[defaultEngine.shortName] ?? nil
```

Comment:
Creating a fake engine key to keep track of this state is a hack; I'd create a separate preference to track it.